Migrate C-I tests from Azure Dev Pipelines to GitHub actions

yantosca opened the pull request:
### Overview
This PR does the following:

1. Removes the Microsoft Azure Dev Pipelines workflow files:
    - `.ci-pipelines/Dockerfile`
    - `.ci-pipelines/build-testing.yml`
    
2. Adds a GitHub Action to run the C-I tests using GCC compiler versions 9, 10, 11, 12, and 13.  (GCC 14 is not yet available on the `ubuntu-latest` 22.04 that is used.)
    - `./github/workflows/run-ci-tests.yml`

3. Adds a GitHub Action to scan other actions for security issues and other problems:
    - `./github/workflows/lint-ci-workflows.yml`

### Advantages:
- We no longer need to maintain an external Azure Dev Pipelines account
- We can run the C-I tests on multiple compilers
- GitHub Actions has enough memory to run all C-I tests, including those for MCM (Azure did not)
- The "Lint" Action will flag any security issues in with the C-I test GitHub action

yantosca committed:
Run C-I tests on GitHub actions instead of Azure Dev Pipelines

.github/workflows/run-ci-tests.yml
- GitHub Actions workflow that will compile KPP and run C-I tests
  using several GNU GCC compiler versions

.ci-pipelines/Dockerfile
.ci-pipelines/build-testing.yml
- Removed these Azure Dev Pipelines workflow config files

CHANGELOG.md
- Updated accordingly

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
Added "Lint" GitHub Action to check other Actions for issues

.github/workflows/lint-ci-workflows.yml
- Added "Lint" GitHub action to check other Actions for security
  and other issues

CHANGELOG.md
- Updated accordingly

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

RolfSander reviewed (approved):
Thanks! It's great to see that GitHub Actions has enough memory for the
MCM tests.

However, I don't know much about C-I tests, so I cannot provide a
detailed review :-( Anyway, if the MCM passes the tests, I'm happy :-)

yantosca committed:
Updated RTD documentation for C-I tests

docs/source/tech_info/06_info_for_kpp_developers.rst
- Updated the documentation for C-I tests to remove references to
  Azure Dev Pipelines and add doc for GitHub Actions.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca commented:
Thanks @RolfSander,   I'm also happy the MCM works!
